operator/pkg/object/objects.go

```diff
@@ -305,20 +306,6 @@ func ParseK8sObjectsFromYAMLManifestFailOption(manifest string, failOnError bool
 	return objects, nil
 }
 
-func removeNonYAMLLines(yms string) string {
-	var b strings.Builder
-	for _, s := range strings.Split(yms, "\n") {
-		if strings.HasPrefix(s, "#") {
-			continue
-		}
-		b.WriteString(s)
-		b.WriteString("\n")
-	}
-
-	// helm charts sometimes emits blank objects with just a "disabled" comment.
-	return strings.TrimSpace(b.String())
```

Comment:
are we sure this cannot change semantics? like ending line of

```yaml
bar: |-
  this line has spaces after it             
```

Reply:
oops this is the deleted code... but I see its just moved

Reply:
Good point, I think this is going to change the semantics as well.